File: NextGreaterElementI/NextGreatEl.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var nextGreaterElement = function (nums1, nums2) {
  return nums1.map((n) => {
    let next = nums2.indexOf(n) + 1;
    while (next < nums2.length && nums2[next] < n) {
      next++;
    }
    return next >= nums2.length ? -1 : nums2[next];
  });
};

File: NextGreaterElementI/NextGreatEl.py
class Solution:
    def nextGreaterElement(self, nums1: List[int], nums2: List[int]) -> List[int]:
        output = []
        for n in nums1:
            idx = nums2.index(n)
            for m in nums2[idx+1:]:
                if m > n:
                    output.append(m)
                    break
            else:
                output.append(-1)
        return output